### PR TITLE
Expose a couple of rigid body properties

### PR DESCRIPTION
**Pull request by andens**
This PR adds component- and editor functionality for friction, rolling friction, spinning friction, coefficient of restitution, linear damping, and angular damping.

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [RigidBody.cpp#L77](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/a3f6868c24754514a164aa165ff74078cfac2333/src/Engine/Component/RigidBody.cpp#L77): The function 'GetPosition' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RigidBody.cpp#L100](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/a3f6868c24754514a164aa165ff74078cfac2333/src/Engine/Component/RigidBody.cpp#L100): The function 'GetOrientation' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [PhysicsManager.cpp#L304](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/a3f6868c24754514a164aa165ff74078cfac2333/src/Engine/Manager/PhysicsManager.cpp#L304): The function 'ForceTransformSync' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Component/RigidBody.hpp [hunk 1]
```diff
@@ -38,6 +38,44 @@ namespace Component {
              */
             ENGINE_API bool IsKinematic() const;
 
+            /// Get the friction coefficient of the rigid body. Not that this
```
Comment (Chainsawkitten) [inline src/Engine/Component/RigidBody.hpp]: Spelling error: not -> note

File: src/Engine/Component/RigidBody.hpp [hunk 2]
```diff
@@ -38,6 +38,44 @@ namespace Component {
...
+             */
+            ENGINE_API float GetRollingFriction() const;
+
+            /// Get the spinning friction coefficient of the rigid body.
+            /**
+             * @return Spinning friction value.
+             */
+            ENGINE_API float GetSpinningFriction() const;
+
+            /// Get the coefficient of restitution the rigid body.
```
Comment (Chainsawkitten) [inline src/Engine/Component/RigidBody.hpp]: Grammar error

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [RigidBody.cpp#L77](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/3609a38f6925b5d98413859c8863c56d04222fe6/src/Engine/Component/RigidBody.cpp#L77): The function 'GetPosition' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RigidBody.cpp#L100](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/3609a38f6925b5d98413859c8863c56d04222fe6/src/Engine/Component/RigidBody.cpp#L100): The function 'GetOrientation' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [PhysicsManager.cpp#L304](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/3609a38f6925b5d98413859c8863c56d04222fe6/src/Engine/Manager/PhysicsManager.cpp#L304): The function 'ForceTransformSync' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [RigidBody.cpp#L77](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/7e54e132b8729485828c88bb4e87018361757397/src/Engine/Component/RigidBody.cpp#L77): The function 'GetPosition' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RigidBody.cpp#L100](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/7e54e132b8729485828c88bb4e87018361757397/src/Engine/Component/RigidBody.cpp#L100): The function 'GetOrientation' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [PhysicsManager.cpp#L304](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/7e54e132b8729485828c88bb4e87018361757397/src/Engine/Manager/PhysicsManager.cpp#L304): The function 'ForceTransformSync' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**